From 120f275f1101ace78b9a54d07562918a8a60047d Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 09 Feb 2022 19:26:01 +0200
Subject: [PATCH] [SW-68803] gaudi3: read fake fuse offset from boot descriptor

This patch add changes to read fake fuse SPI offset from boot
descriptor.

Change-Id: I6d07325ed9458ae89343fe317e5fcdcbeecff388
---

diff --git a/arch/gaudi3/arc/spi_flash.c b/arch/gaudi3/arc/spi_flash.c
index 4ce1a47..99d8659 100644
--- a/arch/gaudi3/arc/spi_flash.c
+++ b/arch/gaudi3/arc/spi_flash.c
@@ -7,6 +7,7 @@
 #include <arcboot.h>
 #include <spi_flash.h>
 #include <efuse.h>
+#include <boot_desc.h>
 
 void spi_flash_init(void)
 {
@@ -37,13 +38,18 @@
 
 void spi_flash_fuse_init(void)
 {
-	uint32_t val;
+	struct boot_desc desc;
+	uint32_t val, idx;
 
 	if (arcboot_rom_global.die_id)
 		return;
 
-	/* TODO: read EFUSE SPI offset from image pointers */
-	arcboot_rom_global.fuse_spi_offset = 0x80000;
+	idx = boot_desc_get_img_desc(HL_FW_BOOT_PTR_SPI, HL_FW_IMG_FAKE_EFUSE,
+					&desc, FLOPS_UPD_NOOP);
+	if (idx == HL_FW_IMG_NOT_FOUND)
+		return;
+
+	arcboot_rom_global.fuse_spi_offset = desc.spi_offset;
 	arcboot_rom_global.fuse_spi = true;
 
 	if (fuse_read(EFUSE_SYSCFG_OFFSET, &val) == -1)
